File: Algo-DS-Python/Search/BinarySearch.py
# Binary Search

import math

def binarySearch(arr, val):
    arr.sort()
    # print(arr)
    start = 0
    end = len(arr) - 1

    while start < end:
        middle = math.floor((start+end)/2)

        if arr[middle] == val:
            print('Find in index', middle)
            return
        
        if arr[middle] < val:
            start = middle + 1
        
        if arr[middle] > val:
            end = middle -1
    


binarySearch([11, 25, 3, 34, 9], 25)

File: DSA-JAVA/Searching/BinarySearch.java
public class BinarySearch {
    public static void main(String[] args) {
        int[] arr = {12, 15, 23, 24, 35, 46, 75, 98};
        int val = 40;

        int start = 0; 
        int end = arr.length - 1;
        
        while(start<=end){
            int middle = (start+end)/2;

            if(arr[middle] == val){
                System.out.println(val+" found at: "+middle);
                break;
            }else if(arr[middle] > val){
                end = middle - 1;
            }else{
                start = middle + 1;
            }
        }

    }
}
